iconic/accounts/managers.py

```diff
@@ -0,0 +1,27 @@
+from django.contrib.auth.base_user import BaseUserManager
+from django.utils.translation import gettext_lazy as _
+from validate_email import validate_email
+
+
+class CustomUserManager(BaseUserManager):#we are defining a custom user model that will handle user authentication for us.
+    def create_user(self, email, password, **extra_fields):
+        if not email:
+            raise ValueError(_('The Email must be set'))
+        email = self.normalize_email(email)
+        if not validate_email(email):
+            raise ValueError(_('Invalid email set'))
+        user = self.model(email=email, **extra_fields)
+        user.set_password(password)
+        user.save()
+        return user
+
+    def create_superuser(self, email, password, **extra_fields): # superuser
```

Comment:
The comment should be in a separate line.

Reply:
It may be worth adding a check for the presence of the email address in the database?